command/transit_import_key.go

```diff
@@ -0,0 +1,171 @@
+)
+
+var (
+	_       cli.Command             = (*TransitImportCommand)(nil)
+	_       cli.CommandAutocomplete = (*TransitImportCommand)(nil)
+	keyPath                         = regexp.MustCompile("^(.*)/keys/([^/]*)$")
+)
+
+type transitImportCommand interface {
+	Flags() *FlagSets
+}
+
+type TransitImportCommand struct {
+	*BaseCommand
+}
+
+func (c *TransitImportCommand) Synopsis() string {
+	return "Import a key into the Transit or Transform secrets engines."
+}
+
+func (c *TransitImportCommand) Help() string {
+	helpText := `
+Usage: vault transit import PATH KEY [options...]
+
+  Using the Transit or Transform key wrapping system, imports key material from
+  the base64 encoded KEY, into a new key whose API path is PATH.  To import a new version
+  into an existing key, use import_version.  The remaining options after KEY (key=value style) are passed
+  on to the transit/transform create key endpoint.  If your system or device natively supports
+  the RSA AES key wrap mechanism, you should use it directly rather than this command. 
+` + c.Flags().Help()
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *TransitImportCommand) Flags() *FlagSets {
+	return c.flagSet(FlagSetHTTP | FlagSetOutputField | FlagSetOutputFormat)
+}
+
+func (c *TransitImportCommand) AutocompleteArgs() complete.Predictor {
+	return nil
+}
+
+func (c *TransitImportCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *TransitImportCommand) Run(args []string) int {
+	return importKey(c.BaseCommand, "import", args)
+}
+
+func importKey(c *BaseCommand, operation string, args []string) int {
+	if len(args) != 2 {
+		c.UI.Error(fmt.Sprintf("Incorrect argument count (expected 2, got %d)", len(args)))
+		return 1
+	}
+
+	client, err := c.Client()
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 2
+	}
+
+	ephemeralAESKey := make([]byte, 32)
+	_, err = rand.Read(ephemeralAESKey)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("failed to generate ephemeral key: %v", err))
+	}
+	parts := keyPath.FindStringSubmatch(args[0])
+	if len(parts) != 3 {
+		c.UI.Error("expected transit path and key name in the form :path:/keys/:name:")
+		return 1
+	}
+	path := parts[1]
+	keyName := parts[2]
+
+	key, err := base64.StdEncoding.DecodeString(args[1])
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("error base64 decoding source key material: %v", err))
+		return 1
+	}
+	// Fetch the wrapping key
+	c.UI.Output("Retrieving transit wrapping key.")
+	wrappingKey, err := fetchWrappingKey(c, client, path)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("failed to fetch wrapping key: %v", err))
+		return 2
+	}
+	c.UI.Output("Wrapping source key with ephemeral key.")
+	wrapKWP, err := subtle.NewKWP(ephemeralAESKey)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("failure building key wrapping key: %v", err))
+		return 2
+	}
+	wrappedTargetKey, err := wrapKWP.Wrap(key)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("failure wrapping source key: %v", err))
+		return 2
```

Comment:
what are your return codes / are they documented somewhere? (in particular 1 vs. 2 vs 3 as failures)

Reply:
Will do.